### PR TITLE
language prefixes

### PR DESCRIPTION
**Pull request by Waino**
- Make the prefix transform configurable via config-config
- Bugfixes to the prefix transform during training
- Bugfixes to the prefix transform during translation
- Avoid dying if a rare bug occurs in the look ahead bucketing

### CODE REVIEW

**Review by TimotheeMickus — COMMENTED** [review]
from what I understand: apparently some tests aren't discovered with unittest, so we should switch the workflow to pytest. I think this workflow change can be added in this PR for simplicity (unless you want to create a separate PR to fix, merge that one first and come to this one to have it merged.

The rest looks fine, aside from one change line 165 in dataloader.py.

File: mammoth/inputters/dataloader.py [hunk 1]
```diff
@@ -139,19 +139,30 @@ def __iter__(self):
...
+                            'StopIteration when trying to pick a bucket in a smart way. '
+                            'Doing something stupid instead. Please check me.'
                         )
-                        current_bucket_idx = smallest_bucket_idx
-                    else:
-                        # there was a larger bucket, shift the index by one
                         current_bucket_idx = next(
                             bucket_idx
-                            for bucket_idx in range(current_bucket_idx, len(self._buckets) + 1)
+                            for bucket_idx in range(len(self._lens))
```
[the remark this answers is not in the document]
Comment (TimotheeMickus) [reply mammoth/inputters/dataloader.py:165]: revert this line, but it's not major (PR #23 is going to nuke this anyways)

[the remark this answers is not in the document]
Comment (Waino) [reply mammoth/inputters/dataloader.py:165]: The diff pairs up the lines in a very unintuitive way. This is easier to read by looking at just the new code:
https://github.com/Helsinki-NLP/mammoth/blob/0b69de60f2d6f7ad5f5e9fb04577eb6a980d169c/mammoth/inputters/dataloader.py#L161

[the remark this answers is not in the document]
Comment (TimotheeMickus) [reply mammoth/inputters/dataloader.py:165]: :upside_down_face:  well I'll be damned. LGTM then, please wait for @onadegibert 's review

[the remark this answers is not in the document]
Comment (Waino) [reply mammoth/inputters/dataloader.py:165]: Do you have a better idea for selecting a guaranteed-to-be-nonempty bucket after the two loops above have failed?

This should be dead code, but if you recall, we did manage to trigger the `StopIteration` when using pathological dummy data. Unless you are sure that you found and fixed that bug, we should have the try/except with some kind of fallback for selecting a bucket. It does not need to be a smart choice.

[the remark this answers is not in the document]
Comment (TimotheeMickus) [reply mammoth/inputters/dataloader.py:165]: PR #20 is going to change the behavior pretty significantly here. 
- On the one hand, the `_spiralling()` pattern should visit all buckets starting from the one we picked for the batch.
- On the other hand, we're looking to support genuine cases of empty LAB's with no available batch (when wrapping around the dataset).

So the try catch is reasonable enough for now.

**Review by onadegibert — APPROVED** [review]
Looks good to me too :)

**Pull request merged**